plugin/specs.py

```diff
@@ -79,3 +82,7 @@ def render_specs(specs_data):
                 outputs.append(output)
 
     return "\n".join(outputs)
+
+
+def add_descriptive_text(data):
+    data["text"] = "\n" + data["text"] if "text" in data else ""
```

Comment:
Rather than adding a newline here, why not add the newlines in each template?

Reply:
because that would leave an extra newline if there were no extra explanatory text present

Reply:
unless there's a way of doing conditionals in python string formatting that I'm unaware of

Reply:
Have you tested that it would visually change the rendered HTML?  Extra newlines in Markdown and HTML _should_ be collapsed, but I'd give it a 50/50 without actually confirming that…

However if that is the case then I think the `paragraph_template` needs to match the other two?

Reply:
> Extra newlines in Markdown and HTML should be collapsed 
That's a very good point which I haven't checked. I will do now.

Reply:
> the paragraph_template needs to match the other two?
there was already a newline there

Reply:
I've added a unit test with the expected markdown and it turns out my first guess wasn't right (yay, testing), so I've altered the paragraph template accordingly and now there's a specification of the expected behaviour